cmake/modules/hunter_check_toolchain_for_define.cmake

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2016 NeroBurner
+# All rights reserved.
+
+include(hunter_test_string_not_empty)
+
+function(hunter_check_toolchain_for_define definition output_var)
+  # the toolchain path must be set
+  hunter_test_string_not_empty("${HUNTER_TOOLCHAIN_ID_PATH}")
+  set(TOOLCHAIN_INFO_FILE "${HUNTER_TOOLCHAIN_ID_PATH}/toolchain.info")
+
+  # read toolchain.info file into memory, each line a entry in a list
+  file(STRINGS "${TOOLCHAIN_INFO_FILE}" toolchain_contents)
+
+  # clean ouptut variable
+  set(definition_found)
+
+  foreach(line ${toolchain_contents})
+    # find the specified define
+    string(REGEX MATCH "^#define ${definition} .*$" match "${line}")
+    if(match)
+      # extract the definition
+      string(REGEX REPLACE "^#define ${definition} " "" definition_found "${match}")
```

Comment:
I was halfway through with an answer about how nice it is to get the defined value, until I've reread your post and understood your point.
`#define A` IS defined, but with the current implementation it would return the same result as if `#define A` was not in `toolchain.info`.

If I just return a boolean (1/0), then what should `#define B 0` return? Because strictly speaking B is defined, but it is defined as false.

If I return the defined value, then empty return values are equal to `#define A`. I could replace the output with true (1) if the value would be empty